misc/backport/backport.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+set -e
+
+BRANCH_NAME=$1
+PR_NUMBER=$2
+
+echo "Backporting PR #$PR_NUMBER to branch $BRANCH_NAME"
+
+# Get the merge commit hash of the pull request
+echo "Fetching merge commit hash of PR #$PR_NUMBER..."
+COMMIT_HASH=$(gh api /repos/"$GITHUB_REPOSITORY"/pulls/"$PR_NUMBER" | jq -r '.merge_commit_sha')
+echo "Merge commit hash: $COMMIT_HASH"
+
+# Get the title of the original pull request
+echo "Fetching title of PR #$PR_NUMBER..."
+ORIGINAL_PR_TITLE=$(gh api /repos/"$GITHUB_REPOSITORY"/pulls/"$PR_NUMBER" | jq -r '.title')
+echo "Original PR title: $ORIGINAL_PR_TITLE"
+
+# Checkout the base branch
+echo "Checking out base branch: $BRANCH_NAME"
+git checkout "$BRANCH_NAME"
+
+# Create a new branch with the PR number and branch name
+NEW_BRANCH="backport-pr-$PR_NUMBER-to-$BRANCH_NAME"
+
+echo "Creating new branch: $NEW_BRANCH"
+git switch -c "$NEW_BRANCH"
```

Comment:
Do we need to upgrage this logic?
e.g. remove remote branch `$NEW_BRANCH` before creating that branch, or pull changes from `$BRANCH_NAME` if `$NEW_BRANCH` exists.

Example:

1. I merged pr1 and pr2 (with changes from pr1). (https://github.com/DmitriyLewen/trivy/pull/90 + https://github.com/DmitriyLewen/trivy/pull/91)
2. I ran the backport for pr2 by mistake and received a conflict warning. (https://github.com/DmitriyLewen/trivy/pull/92)
3. I ran backport for pr1 and merge this backport into `release/vX.Y` branch. (https://github.com/DmitriyLewen/trivy/pull/93)
4. But when i again try to run backport for pr2 - i get error - https://github.com/DmitriyLewen/trivy/actions/runs/9395256621/job/25874336185#step:5:25

Reply:
In that case, I think we should merge `release/vX.Y` into https://github.com/DmitriyLewen/trivy/pull/92 manually. Or we can close https://github.com/DmitriyLewen/trivy/pull/92, delete the branch, and re-run backporting.

> pull changes from $BRANCH_NAME if $NEW_BRANCH exists.

I'm not sure if it's feasible. The conflicted changes are already committed. It sounds like we need to revert.

>remove remote branch $NEW_BRANCH before creating that branch

In this case, we need to force-push the branch. I'm a bit afraid of unexpected accidents (e.g. Important changes are overridden by force push).

Reply:
Okay, let's start with the manual work for cases like this.

We may update this later when we have more experience (if at all necessary).

Reply:
I'll add it to the document.

Reply:
@DmitriyLewen Added https://github.com/aquasecurity/trivy/pull/6781/commits/e1820b441465da5f59710eafcd39b5087dc8cd2b. Please take another look.

Reply:
lgtm